kanidmd/src/lib/filter.rs

```diff
@@ -1164,29 +1214,31 @@ impl FilterResolved {
     fn optimise(&self) -> Self {
         // Most optimisations only matter around or/and terms.
         match self {
-            FilterResolved::Inclusion(f_list) => {
+            FilterResolved::Inclusion(f_list, _) => {
                 // first, optimise all our inner elements
                 let (f_list_inc, mut f_list_new): (Vec<_>, Vec<_>) = f_list
                     .iter()
                     .map(|f_ref| f_ref.optimise())
-                    .partition(|f| matches!(f, FilterResolved::Inclusion(_)));
+                    .partition(|f| matches!(f, FilterResolved::Inclusion(_, _)));
 
                 f_list_inc.into_iter().for_each(|fc| {
-                    if let FilterResolved::Inclusion(mut l) = fc {
+                    if let FilterResolved::Inclusion(mut l, _) = fc {
                         f_list_new.append(&mut l)
                     }
                 });
                 // finally, optimise this list by sorting.
                 f_list_new.sort_unstable();
                 f_list_new.dedup();
-                FilterResolved::Inclusion(f_list_new)
+                // Inclusions are similar to or, so what's our worst case?
+                let sf = f_list_new.last().and_then(|f| f.get_slopeyness_factor());
```

Comment:
Can we replace the closure with:
```rust
<idk type of f>::get_slopeyness_factor
```
I also don't think this would hurt considering that it's impossible to tell the type contained in `f_list_new`s `Vec` just by looking at this function (unless I'm missing something).

Reply:
we need to wait for the sort/dedup first then we can get the max elements, so we could only call the get slopeness after we construction the new or, but to construct the new or we need the slopefactor .... see the problem :)